spec/controllers/tasks_controller_spec.rb

```diff
@@ -3,18 +3,39 @@
 
 RSpec.describe TasksController, type: :controller do
 
+    userObject = {
+            first_name: "TestFirstName", 
+                last_name: "TestLastName", 
+                email:"TestEmail",
+                username: "TestUsername",
+                password: "TestPassword",
+                password_confirmation: "TestPassword"
+            }
+
     describe "POST#create" do
-        it "returns http success" do
-            post :create, params: { task: {:description => "test1"}}
 
-            expect(response).to have_http_status(:success)
+        it "returns a success message" do
+            user = User.create(userObject)
+            allow_any_instance_of(TasksController).to receive(:current_user).and_return(user)
+            post :create, params: { task: {description: "test1"}}
+
+            expect(response.body).to eq("The task was successfully created.")
         end
 
         it "returns an error if the task is not valid" do
+            user = User.create(userObject)
+            allow_any_instance_of(TasksController).to receive(:current_user).and_return(user)
             post :create, params: { task: {:description => nil}}
 
             expect(response.body).to eq("The task couldn't be added. Try again!")
         end
+
+        it "returns an error message if the user is not logged in" do
+            user = User.create(userObject)
+            post :create, params: { task: {description: "test1"}}
+            expect(response.body).to eq("You are not authorized to add this task.")
```

Comment:
Consistent spacing - space should be above the assert.

Reply:
fixed